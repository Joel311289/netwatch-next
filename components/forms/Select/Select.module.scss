.wrapper {
  display: block;
}

.button {
  color: var(--color-text) !important;
  width: fit-content !important;
  height: fit-content !important;

  &.opened {
    background-color: rgb(var(--rgb-background-color-clear-hover) / 8%);
    filter: brightness(87%);
  }
}

.label {
  margin-top: -2px;
  font-weight: 600;
  white-space: normal;
  text-align: left;
}

.arrow {
  font-size: 1.5rem;
  margin-top: 1.5px;
}

.paper {
  margin-top: 7px !important;
  border-radius: 0 !important;
  background-color: transparent !important;
  min-width: 200px !important;
  max-height: 200px !important;
  box-shadow: rgb(0 0 0 / 12%) 0 4px 16px !important;
}

.menu {
  background-color: var(--background-color-primary) !important;
  border: 1px solid rgb(var(--rgb-color-border-portal) / 90%);
}

.item {
  font-family: var(--font-text) !important;
  color: var(--color-text) !important;
  padding: 8px 20px !important;
  font-size: 0.95rem !important;
  font-weight: 500 !important;

  &.selected {
    background-color: rgb(var(--rgb-color-skeleton)) !important;
    opacity: 1;
  }

  &:hover {
    background-color: rgb(var(--rgb-color-skeleton)) !important;
    opacity: 0.7;
  }
}
