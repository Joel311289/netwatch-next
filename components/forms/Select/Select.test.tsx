import { render, screen } from '@testing-library/react';
import Select from '@/components/forms/Select/Select';

describe('Tests Select component', () => {
  test('renders component correctly', () => {
    const { container } = render(
      <Select>
        <Select.Item value="1">Item 1</Select.Item>
        <Select.Item value="2">Item 2</Select.Item>
      </Select>
    );

    expect(container).toMatchSnapshot();
  });

  test('renders placeholder correctly', () => {
    const placeholder = 'Select an item';

    render(
      <Select placeholder={placeholder}>
        <Select.Item value="1">Item 1</Select.Item>
        <Select.Item value="2">Item 2</Select.Item>
      </Select>
    );

    expect(screen.getByText(placeholder)).toBeInTheDocument();
  });

  test('renders selected correctly', () => {
    const placeholder = 'Select an item';

    render(
      <Select placeholder={placeholder} value="1">
        <Select.Item value="1">Item 1</Select.Item>
        <Select.Item value="2">Item 2</Select.Item>
      </Select>
    );

    expect(screen.getByText('Item 1')).toBeInTheDocument();
  });
});
