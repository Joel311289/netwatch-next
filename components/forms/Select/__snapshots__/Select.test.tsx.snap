// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Select component renders component correctly 1`] = `
<div>
  <div
    class="wrapper"
  >
    <button
      class="wrapper medium clear button"
    >
      <div
        class="wrapper"
      >
        <span
          class="label"
        />
        <svg
          class="arrow"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="6 9 12 15 18 9"
          />
        </svg>
        <style>
          
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          gap: 5px;
          
          align-items: center;
        }
      
        </style>
      </div>
    </button>
  </div>
</div>
`;
